sidechain/test/src/sidechain_block_builder.rs

```diff
@@ -111,13 +113,27 @@ impl SidechainBlockBuilder {
 		self
 	}
 
+	/// Calculate the payload of a sidechain block.
+	pub fn block_payload_hash(&self) -> H256 {
+		(
+			self.timestamp,
+			self.parentchain_block_hash,
+			self.signer.public(),
+			&self.signed_top_hashes.clone(),
+			&self.encrypted_payload.clone(),
```

Comment:
We take a reference of the `clone()`? That does not make sense IMO, we should just pass in the reference (using `as_slice()`) without any `clone()`